src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+
+    // Events for transparency to pledgers
+    event RefundBonusDeposited(address, uint256 indexed);
+    event CycleHasClosed(bool indexed, bool indexed);
+    event CycleRefundBonusWithdrawal(address indexed, uint256 indexed);
+    event OwnerWithdrawal(address, uint256);
+    event Test(string, uint256);
+
+    error DataSourceNotInitialized();
+    error FundsMustMatchInputAmount(uint256 input);
+    error AmountIsBelowMinimumPledge(uint256 minAmount);
+    error InvalidPaymentEvent(address caller, uint256 projectId, uint256 value);
+    error CycleHasExpired();
+    error FunctionHasAlreadyBeenCalled();
+    error MustBePledger();
+    error CycleHasNotEndedYet(uint256 endTimestamp);
+    error NoRefundsForSuccessfulCycle();
+    error ContractAlreadyInitialized();
+    error AlreadyWithdrawnRefund();
+    error InvalidRedemptionEvent(address caller, uint256 projectId, uint256 value);
+    error InsufficientFunds();
+
+    /// @notice Indicates if this contract adheres to the specified interface.
+    /// @dev See {IERC165-supportsInterface}.
+    /// @param _interfaceId The ID of the interface to check for adherence to.
+    /// @return A flag indicating if the provided interface ID is supported.
+    function supportsInterface(bytes4 _interfaceId) public view virtual override returns (bool) {
+        return _interfaceId == type(IJBFundingCycleDataSource3_1_1).interfaceId
+            || _interfaceId == type(IJBPayDelegate3_1_1).interfaceId
+            || _interfaceId == type(IJBRedemptionDelegate3_1_1).interfaceId;
+    }
+
+    /// @notice Initializes the Dominant Assurance cloned contract with project details
+    /// and the pertinent Juicebox architecture contracts. Function should be called by the
+    /// Delegate Deployer contract directly after it creates the clone for each new project.
+    /// @param _projectId The ID of the project this contract's functionality applies to.
+    function initialize(
+        uint256 _projectId,
+        uint256 _cycleTarget,
+        uint256 _minimumPledgeAmount,
+        IJBController3_1 _controller,
+        IJBSingleTokenPaymentTerminalStore3_1_1 _paymentTerminalStore
+    ) external {
+        // Stop re-initialization.
+        if (projectId != 0) revert ContractAlreadyInitialized();
+
+        // Store project parameters and ready JB architecture contracts for calling
+        cycleTarget = _cycleTarget;
+        minimumPledgeAmount = _minimumPledgeAmount;
+        projectId = _projectId;
+        controller = _controller;
+        paymentTerminalStore = _paymentTerminalStore;
+        directory = controller.directory();
+        fundAccessConstraintsStore = controller.fundAccessConstraintsStore();
+        address terminalAddr = address(directory.terminalsOf(projectId)[0]);
+        paymentTerminal = IJBSingleTokenPaymentTerminal(terminalAddr);
+        //paymentToken = paymentTerminal.token();
+        (JBFundingCycle memory cycleData,) = controller.currentFundingCycleOf(projectId);
```

Comment:
again, space before this line would aid in readability. and remove the comment on line 135 if it's no longer needed

Reply:
Done.